src/Elliptic/DiscontinuousGalerkin/DgElementArray.hpp

```diff
@@ -3,47 +3,38 @@
 
 #pragma once
 
+#include <cstddef>
+#include <vector>
+
 #include "AlgorithmArray.hpp"
+#include "Domain/Block.hpp"
+#include "Domain/Domain.hpp"
+#include "Domain/ElementId.hpp"
 #include "Domain/ElementIndex.hpp"
 #include "Domain/InitialElementIds.hpp"
 #include "Domain/Tags.hpp"
-#include "IO/Observer/ObservationId.hpp"
-#include "IO/Observer/TypeOfObservation.hpp"
 #include "Parallel/ConstGlobalCache.hpp"
 #include "Parallel/Info.hpp"
-#include "Parallel/Invoke.hpp"
 #include "Parallel/ParallelComponentHelpers.hpp"
 #include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
 
 namespace elliptic {
 /*!
  * \brief The parallel component responsible for managing the DG elements that
  * compose the computational domain
  *
- * This parallel component will perform the following phases:
- *
- * - `Phase::Initialize`: Create the domain and execute
- * `elliptic::dg::Actions::InitializeElement` on all elements.
- * - `Phase::RegisterWithObservers` (optional)
- * - `Phase::Solve`: Execute the actions in `ActionList` on all elements. Repeat
- * until an action terminates the algorithm.
+ * This parallel component will perform the actions specified by the
+ * `PhaseDepActionList`.
  *
  * \note This parallel component is nearly identical to
  * `Evolution/DiscontinuousGalerkin/DgElementArray.hpp` right now, but will
  * likely diverge in the future, for instance to support a multigrid domain.
  *
- * Uses:
- * - Metavariables:
- *   - `domain_creator_tag`
- * - System:
- *   - `volume_dim`
- * - ConstGlobalCache:
- *   - All items required by the actions in `ActionList`
  */
-template <class Metavariables, class PhaseDepActionList,
-          class AddOptionsToDataBox>
+template <class Metavariables, class PhaseDepActionList>
 struct DgElementArray {
-  static constexpr size_t volume_dim = Metavariables::system::volume_dim;
+  static constexpr size_t volume_dim = Metavariables::volume_dim;
```

Comment:
(Just choosing a spot at random): should we remove `volume_dim` from the `System` structs in the future? If so we should create an issue for that so we remember :)

Reply:
yes, issue #1660